### DISCUSSION TITLE
Matrix Block Grouping and Enhanced Drag-and-Drop for Craft CMS

### DISCUSSION THREAD

**Discussion by alexanderbuergin** [craftcms/cms#15554] [Ideas] [answered]
## Request Overview:
I recently proposed a feature enhancement to the Smith plugin by Verbb to improve the usability of Craft CMS Matrix fields. However, I believe that the functionality I suggested would be better suited as a native feature of Craft CMS itself. This enhancement would allow for better content management and a significantly improved user experience.

## The proposed feature includes:
1. Matrix Block Grouping: A native option to group different types of Matrix blocks within the field layout for easier navigation and editing.
2. Drag-and-Drop Between Blocks: The ability to move blocks between different parent blocks (second level) via drag-and-drop, ensuring they belong to the same entry type.

## Proposed Solution:

**Matrix Block Grouping:**
1. Introduce a native feature within Craft CMS that allows users to group Matrix blocks within the field layout.
2. This grouping would visually organize the blocks, making it easier for users to navigate and manage content, especially in more complex setups.

**Grouped Insertion Menu:**
1. When adding a new block, the insertion menu should display the available block types in a grouped format based on their function or category.
2. This organization would maintain clarity and ease of use when managing a large number of block types.

**Drag-and-Drop Between Blocks:**
1. Enable drag-and-drop functionality to move blocks between parent blocks at the second level, ensuring compatibility based on entry type.
2. This would provide flexibility in rearranging content while maintaining structure.

## Benefits:

1. **Improved UX**: Native support for grouping and better block management in Matrix fields would reduce the cognitive load on users and make content creation more efficient.
2. **Streamlined Interface**: A more organized and intuitive interface would benefit all Craft CMS users, particularly those managing complex field setups.
3. **Greater Flexibility**: Drag-and-drop between blocks offers users more control over their content layout, speeding up the workflow.
4. **Consistency Across Tools**: Craft CMS could integrate features similar to those found in popular plugins like MatrixMate and Neo, making the transition between tools smoother for users.

## Conclusion:
While this suggestion was initially directed at Verbb’s Smith plugin, it seems clear that this functionality should be a core feature of Craft CMS. Integrating these enhancements directly into the CMS would greatly improve the user experience for those working heavily with Matrix fields, providing a more organized, efficient, and user-friendly platform.

Thank you for considering this broader feature request.

## Additional context

**From**
<img width="373" alt="355828493-7bfedb49-fae9-4cb7-91dc-eb02386f8187" src="https://github.com/user-attachments/assets/89e711ff-f144-491e-8d57-9f565ffbe297">
<img width="223" alt="355828096-104a7436-102c-4f9f-8364-540ff6117e29" src="https://github.com/user-attachments/assets/929b1e40-8940-4f47-b946-35b5b7a262a4">

**To**
<img width="503" alt="355828102-de13396c-b553-4c9f-99dd-c55f0c5ae355" src="https://github.com/user-attachments/assets/ab40c680-fa42-4d5a-b9e2-a2bab71091f4">
<img width="289" alt="355828554-c82e6810-0e82-4c11-bb30-05fa7c96cf86" src="https://github.com/user-attachments/assets/8aa8e5e5-1c85-4dc8-939f-6ff2bdee2be0">

**Comment by alexanderbuergin** [comment]
See here: https://github.com/verbb/smith/issues/42

**Comment by brandonkelly** [comment, marked as answer]
Thanks for the feedback!

The grouping suggestion is covered by #14179

And moving entries between Matrix fields was requested by #14469
